kernelci/config/__init__.py

```diff
@@ -28,32 +28,32 @@
 import kernelci.config.test
 
 
-def _find_yaml_files(config_path):
+def _iterate_yaml_files(config_path):
     if config_path.endswith('.yaml'):
         yaml_files = [config_path]
     else:
         yaml_files = glob.glob(os.path.join(config_path, "*.yaml"))
-    return yaml_files
-
-
-def validate_yaml(config_path, entries):
-    yaml_files = _find_yaml_files(config_path)
     for yaml_path in yaml_files:
         with open(yaml_path) as yaml_file:
             data = yaml.safe_load(yaml_file)
-            for name, value in data.items():
-                if name not in entries:
-                    continue
-                if hasattr(value, 'update'):
-                    keys = value.keys()
-                elif hasattr(value, 'extend'):
-                    keys = value
-                else:
-                    keys = []
-                err = kernelci.sort_check(keys)
-                if err:
-                    return "Broken order in {} {}: '{}' is before '{}'".format(
-                        yaml_path, name, err[0], err[1])
+            yield(yaml_path, data)
+
+
+def validate_yaml(config_path, entries):
+    for yaml_path, data in _iterate_yaml_files(config_path):
+        for name, value in data.items():
+            if name not in entries:
+                continue
+            if hasattr(value, 'update'):
```

Comment:
Wouldn't it be more clear to use `isinstance(value, dict)` instead?

Reply:
Yes, that was copied from the code that loads the config where it makes more sense to check for the `update` method.  I'll change it here to check the class instead.

Reply:
Fixed